### PR TITLE
Error if port is in use and user configurable port

### PR DESCRIPTION
**Pull request by capitanu**
This small PR adds exception handling for in-use ports when running the server and also allows for port selection in the command.

### CODE REVIEW

**Comment by theoahfeldt** [comment]
The `test/test.mc` file in the repo is not updated to work with the changes from PR [#130](https://github.com/miking-lang/miking/pull/130) in miking, so you get an error when providing this file to the server. You should update this file by removing the outdated alphabet parameters.

File: src/ocaml-server/main.ml [hunk 1]
```diff
@@ -122,32 +130,50 @@ let parse_folder_to_file =
...
+    begin
+      if not !no_file then
+        begin
+
+          ignore @@ Sys.command (String.concat "" ["mi "; !visualize ;" > ../webpage/js/data-source.json "]);
+          Lwt_main.run (Lwt.pick [source_updates ();
+                                  Lwt.catch
+                                    (fun () -> start_server !port ())
+                                    (function
+                                     | _ -> Lwt.fail (PortInUseException (String.concat "" ["Error: Port "; (string_of_int !port); " already in use. Run the same command again with a different port. " ])))])
```
Comment (theoahfeldt) [inline src/ocaml-server/main.ml]: It seems like you are handling all exceptions by raising a self defined `PortInUseException`. But what if `start_server` happens to throw a different error? Why not match against the exception `Unix.Unix_error(Unix.EADDRINUSE, _, _)` and handling it separately, while passing along any other exception?

**Comment by david-broman** [comment]
Not in this PR, but I discovered it now. In the README.md it says

`To install dune, follow the instructions on dune-project.org.`
and then the link is to https://dune-project.org/

What does this link has to do with this project? Solving partial differential equations?

I guess that you mean that you need to install `dune` using the `opam` package manager? I think that the README.md must be updated to reflect this in the correct way. Please do it in this PR.

**Comment by david-broman** [comment]
I have already given this comment, but I do it again. The readme must be updated. It now says 

"command and sourcing your .mc file" is unclear. Sourcing .mc?

**Comment by david-broman** [comment]
Finally, I tested to run the DFA-example. Then I get the following (using macOS 10.15.6)

```
davidbroman@Davids-MacBook-Pro ocaml-server % dune exec ./main.exe ../../examples/dfa.mc 
FILE "/Users/davidbroman/.local/lib/mcore/stdlib/dfa.mc" 27:10-27:21 ERROR: Incorrect application. This is not a function: [(['s','3'])]
Server running, listening on port 3030 for HTTP requests
 http://127.0.0.1:3030
```

I get the error above and then when I test on `http://127.0.0.1:3030` nothing happens.

**Comment by theoahfeldt** [comment]
> Finally, I tested to run the DFA-example. Then I get the following (using macOS 10.15.6)
> 
> ```
> davidbroman@Davids-MacBook-Pro ocaml-server % dune exec ./main.exe ../../examples/dfa.mc 
> FILE "/Users/davidbroman/.local/lib/mcore/stdlib/dfa.mc" 27:10-27:21 ERROR: Incorrect application. This is not a function: [(['s','3'])]
> Server running, listening on port 3030 for HTTP requests
>  http://127.0.0.1:3030
> ```
> 
> I get the error above and then when I test on `http://127.0.0.1:3030` nothing happens.

Ok, so `examples/dfa.mc` and `examples/nfa.mc` are also outdated as of PR [#130](https://github.com/miking-lang/miking/pull/130) in miking

File: src/ocaml-server/main.ml (hunk 1 above)
Comment (capitanu) [reply src/ocaml-server/main.ml]: Fixed in latest commit

**Pull request merged**